pkg/resource/notebook_instance_lifecycle_config/testdata/test_suite.yaml

```diff
@@ -0,0 +1,135 @@
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+          error: nil
+  - name: "Notebook instance lifecycle config readOne tests"
+    description: "Testing the readOne operation"
+    scenarios:
+      - name: "ReadOne=MissingRequiredField"
+        description: "Testing readOne when required field is missing. No API call is made and returns error."
+        given: 
+          desired_state: "v1alpha1/readone/desired/missing_required_field.yaml"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=SuccessClearsConditions"
+        description: Testing a successful reconciliation clears conditions if terminal/recoverable condition were already set to true
+        given:
+          desired_state: "v1alpha1/readone/desired/error_conditions_true.yaml"
+          svc_api:
+            - operation: DescribeNotebookInstanceLifecycleConfigWithContext
+              output_fixture: "sdkapi/describe/success_describe.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/conditions_clear_on_success.yaml"
+      - name: "ReadOne=NotFound"
+        description: "Testing readOne when Describe fails to find the resource on SageMaker"
+        given: 
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeNotebookInstanceLifecycleConfigWithContext
+              error:
+                code: ValidationException
+                message: "Unable to describe Notebook Instance Lifecycle Config unit-testing-notebook-instance-lifecycle-config"
+        invoke: ReadOne
+        expect:
+          error: "resource not found"
+      - name: "ReadOne=Fail"
+        description: "This test checks if the condition is updated if describe fails and readOne returns error"
+        given: 
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeNotebookInstanceLifecycleConfigWithContext
+              error:
+                code: ServiceUnavailable
+                message: "Server is down"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/error_on_describe.yaml"
+          error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+      - name: "ReadOne=AfterCreate"
+        description: "Testing readOne right after create"
+        given:
+          desired_state: "v1alpha1/readone/desired/right_after_create.yaml"
+          svc_api:
+            - operation: DescribeNotebookInstanceLifecycleConfigWithContext
+              output_fixture: "sdkapi/describe/success_describe.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/creating_after_describe.yaml"
+  - name: "Notebook instance lifecycle config update tests"
```

Comment:
Is there a way to check that update completed successfully or is update triggered enough ?

Reply:
I don't think there is since that would be several state transitions.